uint8 idx
float64 length
float64 width
float64 height
float64[9] rot_matrix
float64[3] center